test/env_var_provider_test.exs

```diff
@@ -87,9 +87,9 @@ defmodule EnvVar.ProviderTest do
 
   describe "when the value is a Keyword list" do
     test "it correctly defaults values for numbers, strings, lists, tuples", state do
-      EnvVar.Provider.init(prefix: "beowulf", env_map: state[:complex], enforce: false)
+      config = init_and_load([], prefix: "beowulf", env_map: state[:complex], enforce: false)
```

Comment:
What is beowulf? Should/can we use a custom var?

Reply:
I don't know but it's out of the scope of this PR, which is aimed at making the smallest amount of changes needed to support the new config providers mechanism. We can have a discussion about refactoring this once this gets merged :)

Reply:
This is just text that I put into the tests. It needs a prefix, any prefix for testing. Beowulf happens to be a story I like a lot and when tests fail it's super obvious what is being looked for if it contains that. No reason to make a var for it.